## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 441544`
Commit: `1e8d47984e6d13045d3007d328dddec3e84344cc`

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.mmcv</li>
    <li>python312Packages.mmcv.dist</li>
    <li>python312Packages.mmengine</li>
    <li>python312Packages.mmengine.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.mmcv</li>
    <li>python313Packages.mmcv.dist</li>
    <li>python313Packages.mmengine</li>
    <li>python313Packages.mmengine.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.mmengine</summary>
<pre>    self.pid = os.fork()

tests/test_visualizer/test_visualizer.py::TestVisualizer::test_chain_call
  /private/tmp/nix-build-python3.12-mmengine-0.10.7.drv-0/source/mmengine/visualization/visualizer.py:683: UserWarning: Warning: The circle is out of bounds, the drawn circle may not be in the image
    warnings.warn(

tests/test_visualizer/test_visualizer.py::TestVisualizer::test_draw_bboxes
  /private/tmp/nix-build-python3.12-mmengine-0.10.7.drv-0/source/mmengine/visualization/visualizer.py:831: UserWarning: Warning: The polygon is out of bounds, the drawn polygon may not be in the image
    warnings.warn(

tests/test_visualizer/test_visualizer.py::TestVisualizer::test_init
  /private/tmp/nix-build-python3.12-mmengine-0.10.7.drv-0/source/mmengine/utils/manager.py:113: UserWarning: <class 'mmengine.visualization.visualizer.Visualizer'> instance named of test_save_dir has been created, the method `get_instance` should not accept any other arguments
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_runner/test_amp.py::TestAmp::test_autocast - AssertionError: torch.bfloat16 != torch.float32
FAILED tests/test_runner/test_runner.py::TestRunner::test_test - ValueError: User specified autocast device_type must be cuda or cpu, but got mps
FAILED tests/test_runner/test_runner.py::TestRunner::test_val - ValueError: User specified autocast device_type must be cuda or cpu, but got mps
= 3 failed, 776 passed, 129 skipped, 99 deselected, 247 warnings in 185.76s (0:03:05) =</pre>
</details>
</details>

--- a/pkgs/development/python-modules/mmengine/default.nix
+++ b/pkgs/development/python-modules/mmengine/default.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenv,
   buildPythonPackage,
   fetchFromGitHub,
   fetchpatch,
@@ -117,6 +118,18 @@ buildPythonPackage rec {
 
     # AttributeError: type object 'MagicMock' has no attribute ...
     "tests/test_fileio/test_backends/test_petrel_backend.py::TestPetrelBackend"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    # RuntimeError: attempt to insert nil object from objects[1]
+    "tests/test_visualizer/test_visualizer.py::TestVisualizer::test_draw_featmap"
+    "tests/test_visualizer/test_visualizer.py::TestVisualizer::test_show"
+
+    # AssertionError: torch.bfloat16 != torch.float32
+    "tests/test_runner/test_amp.py::TestAmp::test_autocast"
+
+    # ValueError: User specified autocast device_type must be cuda or cpu, but got mps
+    "tests/test_runner/test_runner.py::TestRunner::test_test"
+    "tests/test_runner/test_runner.py::TestRunner::test_val"
   ];
 
   disabledTests = [
@@ -133,7 +146,15 @@ buildPythonPackage rec {
 
     # AssertionError: os is not <module 'os' (frozen)>
     "test_lazy_module"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    # Fails when max-jobs is set to use fewer processes than cores
+    # for example `AssertionError: assert 14 == 4`
+    "test_setup_multi_processes"
   ];
+
+  # torch.distributed.DistNetworkError: The server socket has failed to bind.
+  __darwinAllowLocalNetworking = true;
 
   meta = {
     description = "Library for training deep learning models based on PyTorch";

